src/main.py

```diff
@@ -1,222 +1,105 @@
-    pprint(type_map)
-    return type_map
-
-
-def get_mapping(filename):
-    with open(filename, 'r') as mapping:
-        data = json.load(mapping)
-        mapping.close()
-        pprint(data)
-        return data
+import psycopg2
+import psycopg2.extras
+from common_utils import get_mapping, get_primary_key, get_type, create_value_map, evaluate_val, disable_triggers, \
+    enable_triggers, create_insert_part
+from db_connections import source, destination, cfg, base_dir, logger
+from argument_list import src_dest_table_names, src_schema_name, dest_schema_name
 
 
-def migrate(destination_schema, destination_table, schema_name, table_name):
-    file = "%s/%s.json" % (base_dir, table_name)
+def create_batch_insert(destination_schema, destination_table, schema_name, table_name):
+    file = "%s/mappings/%s.json" % (base_dir, table_name)
     mapping = get_mapping(file)
-    col_type = get_type(destination_schema, destination_table)
+    col_type_dest = get_type(destination_schema, destination_table)
+    pk = get_primary_key(table_name)
     lst = list()
     for key in mapping:
-        lst.append(key)
+        new_key = key.join('""')
+        new_key = new_key.replace('\'', '')
+        lst.append(new_key)
+    if pk:
+        lst.append('ROW_NUMBER() OVER(ORDER BY %s)' % pk)
     select_str = ",".join(lst)
-    sql_select = 'select ' + select_str + ' from %s.%s' % (schema_name, table_name)
+    sql_count = 'select count(*) from %s.%s' % (schema_name, table_name)
     cur = source.cursor(cursor_factory=psycopg2.extras.DictCursor)
-    cur.execute(sql_select)
-    row = cur.fetchone()
-
-    while row is not None:
-        value_map = dict()
-        for key in mapping:
-            value_map[mapping[key]] = row[key]
-        create_insert(value_map, col_type, destination_schema, destination_table)
-        row = cur.fetchone()
+    cur.execute(sql_count)
+    results = cur.fetchone()
+    no_rows = results['count']
+    logger.info('No of Rows : ' + '%s' % no_rows)
     cur.close()
 
-
-def create_insert(value_map, col_type, schema_name, table_name):
-    key_lst = list()
-    value_lst = list()
-    query = "INSERT INTO %s.%s(%s) VALUES (%s)"
-    values = ()
-    for key in col_type:
-        key_lst.append(key)
-        if key in value_map.keys():
-            values = values + (evaluate_val(col_type, key, value_map[key]),)
-            value_lst.append('%s')
+    batch_sz = int(cfg['source']['batch'])
+    itr = 1
+    if no_rows > batch_sz:
+        itr = int((no_rows / batch_sz) + 1)
+    logger.info('No of iterations : ' + '%s' % str(itr))
+
+    sql, key_lst = create_insert_part(destination_schema, destination_table, col_type_dest, mapping)
+    new_key_lst = []
+    for data in key_lst:
+        a = data.strip('"')
+        new_key_lst.append(a)
+
+    for i in range(itr):
+        offset = int(i * batch_sz)
+        if pk:
+            sql_select = 'select ' + select_str + ' from %s.%s order by ROW_NUMBER LIMIT %s OFFSET %s' \
```

Comment:
Any reason why we are using string concatenation instead of  string formatting here?